modules/pubmaticBidAdapter.js

```diff
@@ -737,6 +739,34 @@ function _addFloorFromFloorModule(impObj, bid) {
   impObj.bidfloor = ((!isNaN(bidFloor) && bidFloor > 0) ? bidFloor : UNDEFINED);
 }
 
+function _populateSegmentDataIfAvailable(fpdUserData, segmentData) {
+  // fpdUserData - data set from global and bidder level configs
+  // segmentData - data set from permutive
+  var dataMerged = false;
+  if (!fpdUserData.data && segmentData.length > 0) {
+    fpdUserData.data = [{
+      name: PERMUTIVE_SEGMENT_NAME,
+      segment: []
+    }]
+  }
+  for (var id in fpdUserData.data) {
+    if (fpdUserData.data[id].name === PERMUTIVE_SEGMENT_NAME) {
+      var mergedArr = utils.removeDuplicatesFromObjectArray(fpdUserData.data[id].segment.concat(segmentData), 'id');
```

Comment:
can we avoid this function with arrayName.filter(utils.uniques)?
Refer: https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L612

Reply:
this array is an arry of objects of the format {id: '1'}. so a normal array filtering function wont work.